.github/workflows/aws.yml

```diff
@@ -0,0 +1,128 @@
+# Basic workflow with Actions
+
+name: AWS Cloud Database Test Execution
+on:
+  # Triggers the workflow on push & pull request events for the main branch. Also allows for manual triggers
+  # push:
+  #   branches: [ main ]
+  # pull_request:
+  #   branches: [ main ]
+  # schedule:
+  #   - cron: '0 12 * * *' # Execute every day at noon
+  workflow_dispatch:
+
+
+jobs:
+  setup:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '1.8.0'
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          # terraform_version: 0.13.0:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+
+      - name: Terraform Format
+        working-directory: src/test/resources/terraform
+        id: fmt
+        run: terraform fmt -check
+
+      - name: Terraform Init
+        working-directory: src/test/resources/terraform
+        id: init
+        run: terraform init
+      
+      - name: Terraform Validate
+        working-directory: src/test/resources/terraform
+        id: validate
+        run: terraform validate -no-color
+
+      - name: Terraform Plan
+        working-directory: src/test/resources/terraform
+        id: plan
+        if: github.event_name == 'pull_request'
+        run: terraform plan -no-color
```

Comment:
It is [generally recommended](https://www.terraform.io/docs/cli/commands/plan.html) to use the `-out=FILE` flag during automation so the later apply step uses this generated plan file. Is there any particular reason not to do this here? I think it might be a good idea especially since there are a couple of steps below that happen before the apply phase.

Reply:
Thanks @mcred !  I was following this:
https://learn.hashicorp.com/tutorials/terraform/github-actions?in=terraform/automation

Perhaps the `-out=FILE` is not required because I'm using TF Cloud?